inject/src/main/java/io/micronaut/context/AbstractInitializableBeanDefinitionReference.java

```diff
@@ -67,6 +68,26 @@ public AbstractInitializableBeanDefinitionReference(String beanTypeName, String
                                                         boolean isPrimary, boolean isContextScope, boolean isConditional,
                                                         boolean isContainerType, boolean isSingleton, boolean isConfigurationProperties,
                                                         boolean hasExposedTypes, boolean requiresMethodProcessing) {
+        this(beanTypeName, beanDefinitionTypeName, annotationMetadata, isPrimary, isContextScope, isConditional, isContainerType, isSingleton, isConfigurationProperties, hasExposedTypes, requiresMethodProcessing, false);
+    }
+
+    /**
+     * @param beanTypeName              The bean type name
+     * @param beanDefinitionTypeName    The bean definition type name
+     * @param annotationMetadata        The annotationMetadata
+     * @param isPrimary                 Is primary bean?
+     * @param isContextScope            Is context scope?
+     * @param isConditional             Is conditional? = No @Requires
+     * @param isContainerType           Is container type?
+     * @param isSingleton               Is singleton?
+     * @param isConfigurationProperties Is configuration properties?
+     * @param hasExposedTypes           Has exposed types?
+     * @param requiresMethodProcessing  Is requires method processing?
+     */
+    protected AbstractInitializableBeanDefinitionReference(String beanTypeName, String beanDefinitionTypeName, AnnotationMetadata annotationMetadata,
```

Comment:
Can you introduce a new object to represent the internal state which would be generated and passed via the constructor? Something like:

```java
class Data {
//...
     private final boolean isConfigurationProperties;
    private final boolean hasExposedTypes;
    private final boolean requiresMethodProcessing;
    private final boolean isProxiedBean;
//...
}

class AbstractInitializableBeanDefinitionReference {

    private final String beanTypeName;
    private final String beanDefinitionTypeName;
    private final AnnotationMetadata annotationMetadata;
    private final Data data;

    protected AbstractInitializableBeanDefinitionReference(Data data) {
          this.data = data;
    }

    public boolean isProxiedBean() {
        return data.isProxiedBean;
    }
}
```

I'm planing to add something for CDI, and adding/modifying the constructor IMHO is a pain.